vector/src/main/scala/geotrellis/vector/json/JsonFeatureCollectionMap.scala

```diff
@@ -0,0 +1,113 @@
+package geotrellis.vector.json
+
+import spray.json._
+
+import geotrellis.vector._
+import scala.util.{Try, Success, Failure}
+import FeatureFormats._
+import DefaultJsonProtocol._
+
+/**
+ * Accumulates GeoJson from Feature class instances and implements a Map keyed on geojson feature IDs.
+ *
+ * During serialization:
+ * Each individual feature is parametrized on a class we need to accumulate geoJson per
+ * instance of an object in order to use implicit scope resolution in finding the correct format.
+ *
+ * Features may be added using the .add, addAll methods, they are buffered as JsValues until .toJson is called
+ *
+ * During deserialization:
+ * This object is instantiated with list of JsValues representing features.
+ * It may be queried using .getAll[F <: Feature[_] ] method.
+ *
+ * It aggregates feature objects with data member still encoded in json
+ */
+class JsonFeatureCollectionMap(features: List[JsValue] = Nil) {
+  private var buffer = features
+
+  //-- Used for Serialization
+  def add[D: JsonWriter](featureMap: (String, Feature[D])) =
+    buffer = writeFeatureJsonWithID[D](featureMap) :: buffer
+  def +=[D: JsonWriter](featureMap: (String, Feature[D])) = add(featureMap)
+
+  def addAll[D: JsonWriter](featureMaps: Seq[(String, Feature[D])]) =
+    featureMaps.foreach{ f => buffer = writeFeatureJsonWithID[D](f) :: buffer }
+  def ++=[D: JsonWriter](featureMaps: Seq[(String, Feature[D])]) = addAll(featureMaps)
+
+  def add(geometry: Geometry) =
+    buffer = geometry.toJson :: buffer
+  def +=(id: String, geometry: Geometry) = add(geometry)
+
+  def addAll(geometries: Seq[Geometry]) =
+    geometries.foreach{ f => add(f) }
+  def ++=(geometries: Seq[Geometry]) = addAll(geometries)
+
+  def toJson: JsValue =
+    JsObject(
+      "type" -> JsString("FeatureCollection"),
+      "features" -> JsArray(buffer)
+    )
+
+  // This helper function is called below to grab the ID field for Map keys
+  private def getFeatureID(js: JsValue): String = {
+    Try(js.asJsObject.getFields("id").head.convertTo[String]) match {
```

Comment:
I'm feeling Try hate. The normal pattern is to match on Seq from .getFields; Seq() => "id" does not exist